src/commons/asBaseComponent.tsx

```diff
@@ -33,7 +33,7 @@ function asBaseComponent<PROPS, STATICS = {}>(WrappedComponent: React.ComponentT
     };
 
     static getDerivedStateFromError(error: any) {
-      UIComponent.defaultProps?.onError(error, WrappedComponent.defaultProps);
+      UIComponent.defaultProps?.onError && UIComponent.defaultProps.onError(error, WrappedComponent.defaultProps);
```

Comment:
You can also invoke using optional chaining (: 
```suggestion
      UIComponent.defaultProps?.onError?.(error, WrappedComponent.defaultProps);
```